yateto/gemm_configuration.py

```diff
@@ -178,15 +194,41 @@ def preference(self, m, n, k, sparseA, sparseB, transA, transB, alpha, beta):
   def blockSize(self, m, n, k):
     return dict()
 
+
+class GemmForge(CodeGenerator):
+  def __init__(self, arch, threshold: int = 256):
+    super().__init__('', ['gemmgen_aux.h'], '', arch)
+    self._threshold = threshold
+
+  def _is_arch_supported(self):
+    return self._arch.name.lower() in {'nvidia'}
+
+  def supported(self, m, n, k, sparseA, sparseB, transA, transB, alpha, beta, platform):
+    if self._is_arch_supported():
+      if not (sparseA and sparseB):
+        if platform == 'gpu':
+          return True
```

Comment:
should be return self._is_arch_supported() and not (sparseA and sparseB) and platform == 'gpu'
But: do you actually support the case where one matrix is sparse?

Reply:
Here is a mistake in de Morgan's Law.
```
 if not (sparseA and sparseB):
```
It should be:
```
 if not (sparseA or sparseB):
```


> should be return self._is_arch_supported() and ... and ...

I know what you mean, I don't like it. It is difficult to read and understand. I would agree to merge two last inner if-statements
```
if self._is_arch_supported():
  if platform == 'gpu' and not (sparseA or sparseB):
    return True
```